tests/integration/events.bats

```diff
@@ -111,3 +111,56 @@ function teardown() {
   run eval "grep -q 'test_busybox' events.log"
   [ "$status" -eq 0 ]
 }
+
+@test "events oom " {
+  # XXX: currently cgroups require root containers.
+  requires root
+  init_cgroup_paths
+
+  # Set some initial known values
+  if [ "$CGROUP_UNIFIED" == "yes" ]; then
+  DATA=$(cat <<EOF
+  "memory": {
+    "limit": 33554433,
+    "kernel": 33554433,
+    "swappiness": 0,
+    "swap": 33554434
+  },
+EOF
+  )
+  else
+  DATA=$(cat <<EOF
+  "memory": {
+    "limit": 33554433,
+    "kernel": 33554433,
+    "swappiness": 0,
+    "swap": 0
+  },
+EOF
+  )
+  fi
+  DATA=$(echo ${DATA} | sed 's/\n/\\n/g')
+  sed -i "s/\(\"resources\": {\)/\1\n${DATA}/" ${BUSYBOX_BUNDLE}/config.json
+
+  # run busybox detached
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox
+  [ "$status" -eq 0 ]
+
+  # spawn two sub processes (shells)
+  # the first sub process is an event logger that sends stats events to events.log
+  # the second sub process exec a memory hog process to cause a oom condition
+  # and waits for an oom event that includes test_busybox
+  # then kills the test_busybox container which causes the event logger to exit
+  (__runc events test_busybox > events.log) &
+  (
+    retry 10 1 eval "grep -q 'test_busybox' events.log"
+    __runc exec -d test_busybox sh -c 'VAR=$(seq 1 100000000)'
+    retry 10 1 eval "grep -q 'oom' events.log"
+    __runc delete -f test_busybox
+  ) &
+  wait # wait for the above sub shells to finish
```

Comment:
Maybe add a small timeout in case runc won't ever exit?
`timeout 10 wait`

Reply:
Once  #2370 merged, I'll update it together with some other changes.

Reply:
@lifubang #2370 is merged

Reply:
@kolyshkin `timeout` and `wait` can't be used together.

Reply:
Yes, you're right.

I am still concerned that this case might hang forever. Maybe we can run both `__runc` invocations with a timeout then. Something big, say 1 minute. Obviously, if OOM has not happened within a minute, the test has failed.

Reply:
Travis CI will return an error if there is no output after 150 seconds.
I met this error before I disable swap.

If you worry about this, I can write simple C program to ensure causing oom kill.

Reply:
@kolyshkin use `sh -c 'test=$(dd if=/dev/urandom ibs=5120k)'` to ensure oom kill.
Because the memory limit is 4M.